src/scripts/items/EvolutionStone.ts

```diff
@@ -14,33 +14,40 @@ class EvolutionStone extends Item {
         player.gainItem(GameConstants.StoneType[this.type], n)
     }
 
-
-
     public use(pokemon?:string) {
         let shiny = PokemonFactory.generateShiny(GameConstants.SHINY_CHANCE_STONE);
-        if(pokemon == "Eevee") {
-            switch (this.type) {
+        let evolution = EvolutionStone.computeEvolution(this.type, pokemon);
+        player.capturePokemon(evolution, shiny, false);
+        return shiny;
```

Comment:
Why return the boolean `shiny`?

Reply:
It returns `shiny` because `ItemHandler.useStones` needs it to determine when to stop using the stones if player selects "Until Shiny". But the implementation of `ItemHandler.useStones` is a bit flawed, which I can fix in a separate PR.